src/memo/memo.ml

```diff
@@ -688,3 +688,22 @@ end
 module Async = struct
   type nonrec ('i, 'o) t = ('i, 'o, 'i -> 'o Fiber.t) t
 end
+
+let lazy_ (type a) f =
+  let module Output = struct
+    type t = a
+    let to_sexp _ = Sexp.Atom "<opaque>"
+    let equal = (==)
+  end
+  in
+  let memo =
+    create "lazy"
+      ~doc:("a lazy value constructed at \n" ^
```

Comment:
Isn't it enough to use `__POS__` here? I'm not such a fan of adding callstacks to what we call a documentation field. Another option would be to have a `callstack` or `loc` optional argument to `Memo.create`.

Reply:
I think this backtrace will be useful for debugging, but it's not used for now and it's just one line so I'm removing it and we can bring it back when we actually need it. The form we need will be more clear then (and the usefulness will be easier to assess).

I'm adding an "id" which will at least appear in the reported dependency cycles.